File: CS/SpreadsheetDocServerPivotAPI/CodeUtils/SpreadsheetExampleEvaluatorByTimer.cs
namespace SpreadsheetDocServerPivotAPI
{
    #region SpreadsheetExampleEvaluatorByTimer
    public class SpreadsheetExampleEvaluatorByTimer : ExampleEvaluatorByTimer
    {
        public SpreadsheetExampleEvaluatorByTimer()
            : base()
        {
        }

        protected override ExampleCodeEvaluator GetExampleCodeEvaluator(ExampleLanguage language)
        {
            if (language == ExampleLanguage.VB)
                return new SpreadsheetVbExampleCodeEvaluator();
            return new SpreadsheetCSExampleCodeEvaluator();
        }
    }
    #endregion
}

File: VB/SpreadsheetDocServerPivotAPI/CodeUtils/SpreadsheetExampleEvaluatorByTimer.vb
Namespace SpreadsheetDocServerPivotAPI
    #Region "SpreadsheetExampleEvaluatorByTimer"
    Public Class SpreadsheetExampleEvaluatorByTimer
        Inherits ExampleEvaluatorByTimer

        Public Sub New()
            MyBase.New()
        End Sub

        Protected Overrides Function GetExampleCodeEvaluator(ByVal language As ExampleLanguage) As ExampleCodeEvaluator
            If language = ExampleLanguage.VB Then
                Return New SpreadsheetVbExampleCodeEvaluator()
            End If
            Return New SpreadsheetCSExampleCodeEvaluator()
        End Function
    End Class
    #End Region
End Namespace
